pkg/utils/selector.go

```diff
@@ -78,20 +77,20 @@ func SelectAndFilterPods(ctx context.Context, c client.Client, r client.Reader,
 // SelectPods returns the list of pods that are available for pod chaos action.
 // It returns all pods that match the configured label, annotation and namespace selectors.
 // If pods are specifically specified by `selector.Pods`, it just returns the selector.Pods.
-func SelectPods(ctx context.Context, c client.Client, r client.Reader, selector v1alpha1.SelectorSpec) ([]v1.Pod, error) {
+func SelectPods(ctx context.Context, c client.Client, r client.Reader, selector v1alpha1.SelectorSpec, clusterScoped bool, targetNamespace string, allowedNamespaces, ignoredNamespaces string) ([]v1.Pod, error) {
```

Comment:
It's kind of ugly, should we rethink how we do management configuration? Maybe it's time for a round of refactoring?

Reply:
Yes. I think we should build a `selector` thing, that contains clients, configurations as its state.

I rememeber we have a talk about this on https://github.com/chaos-mesh/chaos-mesh/issues/886.

Now we hae rfc repo, how about creating a RFC about the selector?

Reply:
Yes, from my view, I think we'd better to call `SelectAndFilterPods`(it will call `SelectPods`) from the common reconciler layer, rather than each chaos endpoint's `Apply` method, therefore, we could consume the controllerCfg and don't need to pass the controller cfg into the util methods.

But it would need a lot of refactoring, so I wonder if we do this job in this PR, or put them into another PR.

Reply:
BTW, current implementation would cause lint to produce waring:
```
linting
~/go/bin/revive -formatter friendly -config revive.toml $(go list ./... | grep -vE "chaos-mesh/test|pkg/ptrace|zz_generated|vendor")
  ⚠  https://revive.run/r#flag-parameter  parameter 'clusterScoped' seems to be a control flag, avoid control coupling
 ~/go/src/github.com/chaos-mesh/pkg/utils/selector.go:80:16

  ⚠  https://revive.run/r#flag-parameter  parameter 'clusterScoped' seems to be a control flag, avoid control coupling
  ~/go/src/github.com/chaos-mesh/pkg/utils/selector.go:80:16

  ⚠  https://revive.run/r#flag-parameter  parameter 'clusterScoped' seems to be a control flag, avoid control coupling
  ~/go/src/github.com/chaos-mesh/pkg/utils/selector.go:80:16

⚠ 3 problems (0 errors, 3 warnings)

Warnings:
  3  flag-parameter
```

Reply:
Yes, a workaround for this warning should be that passing the cfg as the `SelectPods` parameter, but I think the best solution should be that refactoring the `SelectAndFilterPods ` and calling it from the common reconciler layer.